SLM metadata incorrects skips parsing operation mode

In `SnapshotLifecycleMetadata` the XContent parser skips adding a parser for the operation mode. This causes the following exception on startup:

```
[2019-09-09T09:34:57,864][ERROR][o.e.g.GatewayMetaState   ] [perelandra] failed to read or upgrade local state, exiting...
org.elasticsearch.ElasticsearchException: java.io.IOException: failed to read /home/hinmanm/elasticsearch-7.4.0-SNAPSHOT/data/nodes/0/_state/global-22.st
	at org.elasticsearch.ExceptionsHelper.maybeThrowRuntimeAndSuppress(ExceptionsHelper.java:167) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:414) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaStateService.loadFullState(MetaStateService.java:81) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.upgradeMetaData(GatewayMetaState.java:141) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:95) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:485) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:255) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) [elasticsearch-cli-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cli.Command.main(Command.java:90) [elasticsearch-cli-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:115) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) [elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
Caused by: java.io.IOException: failed to read /home/hinmanm/elasticsearch-7.4.0-SNAPSHOT/data/nodes/0/_state/global-22.st
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:408) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 15 more
Caused by: org.elasticsearch.common.xcontent.XContentParseException: [-1:24494] [snapshot_lifecycle] failed to parse field [policies]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:386) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:396) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:238) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.xpack.ilm.IndexLifecycle.lambda$getNamedXContent$3(IndexLifecycle.java:186) ~[?:?]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:63) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:141) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:385) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$Builder.fromXContent(MetaData.java:1403) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1448) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1439) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:302) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:404) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 15 more
Caused by: org.elasticsearch.common.xcontent.XContentParseException: Failed to build [snapshot_lifecycle] after last required field arrived
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.buildTarget(ConstructingObjectParser.java:476) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.constructorArg(ConstructingObjectParser.java:402) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.access$900(ConstructingObjectParser.java:350) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.lambda$declareNamedObjects$5(ConstructingObjectParser.java:288) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.lambda$declareNamedObjects$7(ObjectParser.java:345) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:384) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:396) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:238) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.xpack.ilm.IndexLifecycle.lambda$getNamedXContent$3(IndexLifecycle.java:186) ~[?:?]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:63) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:141) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:385) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$Builder.fromXContent(MetaData.java:1403) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1448) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1439) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:302) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:404) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 15 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at org.elasticsearch.xpack.core.slm.SnapshotLifecycleMetadata.lambda$static$1(SnapshotLifecycleMetadata.java:51) ~[?:?]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.lambda$new$2(ConstructingObjectParser.java:132) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.buildTarget(ConstructingObjectParser.java:463) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.constructorArg(ConstructingObjectParser.java:402) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.access$900(ConstructingObjectParser.java:350) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.lambda$declareNamedObjects$5(ConstructingObjectParser.java:288) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.lambda$declareNamedObjects$7(ObjectParser.java:345) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:384) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:396) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:238) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.xpack.ilm.IndexLifecycle.lambda$getNamedXContent$3(IndexLifecycle.java:186) ~[?:?]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:63) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:141) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:385) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$Builder.fromXContent(MetaData.java:1403) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1448) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1439) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:302) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:404) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 15 more
[2019-09-09T09:34:57,907][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [perelandra] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: ElasticsearchException[java.io.IOException: failed to read /home/hinmanm/elasticsearch-7.4.0-SNAPSHOT/data/nodes/0/_state/global-22.st]; nested: IOException[failed to read /home/hinmanm/elasticsearch-7.4.0-SNAPSHOT/data/nodes/0/_state/global-22.st]; nested: XContentParseException[[-1:24494] [snapshot_lifecycle] failed to parse field [policies]]; nested: XContentParseException[Failed to build [snapshot_lifecycle] after last required field arrived]; nested: ArrayIndexOutOfBoundsException[Index 1 out of bounds for length 1];
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:163) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:150) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) ~[elasticsearch-cli-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:115) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
Caused by: org.elasticsearch.ElasticsearchException: java.io.IOException: failed to read /home/hinmanm/elasticsearch-7.4.0-SNAPSHOT/data/nodes/0/_state/global-22.st
	at org.elasticsearch.ExceptionsHelper.maybeThrowRuntimeAndSuppress(ExceptionsHelper.java:167) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:414) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaStateService.loadFullState(MetaStateService.java:81) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.upgradeMetaData(GatewayMetaState.java:141) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:95) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:485) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:255) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 6 more
Caused by: java.io.IOException: failed to read /home/hinmanm/elasticsearch-7.4.0-SNAPSHOT/data/nodes/0/_state/global-22.st
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:408) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaStateService.loadFullState(MetaStateService.java:81) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.upgradeMetaData(GatewayMetaState.java:141) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:95) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:485) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:255) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 6 more
Caused by: org.elasticsearch.common.xcontent.XContentParseException: [-1:24494] [snapshot_lifecycle] failed to parse field [policies]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:386) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:396) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:238) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.xpack.ilm.IndexLifecycle.lambda$getNamedXContent$3(IndexLifecycle.java:186) ~[?:?]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:63) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:141) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:385) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$Builder.fromXContent(MetaData.java:1403) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1448) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1439) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:302) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:404) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaStateService.loadFullState(MetaStateService.java:81) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.upgradeMetaData(GatewayMetaState.java:141) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:95) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:485) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:255) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 6 more
Caused by: org.elasticsearch.common.xcontent.XContentParseException: Failed to build [snapshot_lifecycle] after last required field arrived
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.buildTarget(ConstructingObjectParser.java:476) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.constructorArg(ConstructingObjectParser.java:402) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.access$900(ConstructingObjectParser.java:350) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.lambda$declareNamedObjects$5(ConstructingObjectParser.java:288) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.lambda$declareNamedObjects$7(ObjectParser.java:345) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:384) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:396) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:238) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.xpack.ilm.IndexLifecycle.lambda$getNamedXContent$3(IndexLifecycle.java:186) ~[?:?]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:63) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:141) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:385) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$Builder.fromXContent(MetaData.java:1403) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1448) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1439) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:302) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:404) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaStateService.loadFullState(MetaStateService.java:81) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.upgradeMetaData(GatewayMetaState.java:141) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:95) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:485) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:255) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 6 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at org.elasticsearch.xpack.core.slm.SnapshotLifecycleMetadata.lambda$static$1(SnapshotLifecycleMetadata.java:51) ~[?:?]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.lambda$new$2(ConstructingObjectParser.java:132) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.buildTarget(ConstructingObjectParser.java:463) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.constructorArg(ConstructingObjectParser.java:402) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser$Target.access$900(ConstructingObjectParser.java:350) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.lambda$declareNamedObjects$5(ConstructingObjectParser.java:288) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.lambda$declareNamedObjects$7(ObjectParser.java:345) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseValue(ObjectParser.java:384) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parseSub(ObjectParser.java:396) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ObjectParser.parse(ObjectParser.java:238) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.ConstructingObjectParser.parse(ConstructingObjectParser.java:169) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.xpack.ilm.IndexLifecycle.lambda$getNamedXContent$3(IndexLifecycle.java:186) ~[?:?]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:63) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:141) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:385) ~[elasticsearch-x-content-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$Builder.fromXContent(MetaData.java:1403) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1448) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.cluster.metadata.MetaData$1.fromXContent(MetaData.java:1439) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.read(MetaDataStateFormat.java:302) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaDataStateFormat.loadGeneration(MetaDataStateFormat.java:404) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.MetaStateService.loadFullState(MetaStateService.java:81) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.upgradeMetaData(GatewayMetaState.java:141) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:95) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:485) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.node.Node.<init>(Node.java:255) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:159) ~[elasticsearch-7.4.0-SNAPSHOT.jar:7.4.0-SNAPSHOT]
	... 6 more
```

And prevents the node from starting.

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecycleMetadata.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecycleMetadata.java
@@ -56,6 +56,7 @@ public class SnapshotLifecycleMetadata implements XPackMetaDataCustom {
             v -> {
                 throw new IllegalArgumentException("ordered " + POLICIES_FIELD.getPreferredName() + " are not supported");
             }, POLICIES_FIELD);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), OPERATION_MODE_FIELD);
     }
 
     private final Map<String, SnapshotLifecyclePolicyMetadata> snapshotConfigurations;

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/slm/SnapshotLifecycleMetadataTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/slm/SnapshotLifecycleMetadataTests.java
@@ -1,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.slm;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+import org.elasticsearch.xpack.core.ilm.OperationMode;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SnapshotLifecycleMetadataTests extends AbstractSerializingTestCase<SnapshotLifecycleMetadata> {
+    @Override
+    protected SnapshotLifecycleMetadata doParseInstance(XContentParser parser) throws IOException {
+        return SnapshotLifecycleMetadata.PARSER.apply(parser, null);
+    }
+
+    @Override
+    protected SnapshotLifecycleMetadata createTestInstance() {
+        int policyCount = randomIntBetween(0, 3);
+        Map<String, SnapshotLifecyclePolicyMetadata> policies = new HashMap<>(policyCount);
+        for (int i = 0; i < policyCount; i++) {
+            String id = "policy-" + randomAlphaOfLength(3);
+            policies.put(id, SnapshotLifecyclePolicyMetadataTests.createRandomPolicyMetadata(id));
+        }
+        return new SnapshotLifecycleMetadata(policies, randomFrom(OperationMode.values()));
+    }
+
+    @Override
+    protected Writeable.Reader<SnapshotLifecycleMetadata> instanceReader() {
+        return SnapshotLifecycleMetadata::new;
+    }
+}

